vmm/src/landlock.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright © 2024 Microsoft Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use landlock::{
+    path_beneath_rules, Access, AccessFs, BitFlags, PathFdError, Ruleset, RulesetAttr,
+    RulesetCreated, RulesetCreatedAttr, RulesetError, ABI,
+};
+use std::io::Error as IoError;
+use std::path::PathBuf;
+use thiserror::Error;
+
+use crate::vm_config::LandlockConfig;
+
+pub const READ: u8 = 1 << 0;
+pub const WRITE: u8 = 1 << 1;
+pub const EXECUTE: u8 = 1 << 2;
+
+#[derive(Debug, Error)]
+pub enum LandlockError {
```

Comment:
Do we really need this new error type? `RulesetError` should already contain the required information/types.

Reply:
For the most part, the errors here map to ones in `RulesetError`. The only case that doesn't have an error in `RulesetError` is when the ruleset is empty. Such a case doesn't map to a proper error in RulesetError. 

If you can add such an error case, I will drop LandlockError here.

Reply:
> The only case that doesn't have an error in `RulesetError` is when the ruleset is empty. Such a case doesn't map to a proper error in RulesetError.
> 
> If you can add such an error case, I will drop LandlockError here.

An empty ruleset is not an error because it can be used to just drop a set of access rights (without exception), for instance when only using inherited file descriptors.